### PR TITLE
fix: handle expiration times as Duration

### PR DESCRIPTION
**Pull request by hko-s**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
why?

**Comment by hko-s** [comment]
The values of the two expiration time subpackets are defined as duration in seconds, not as absolute dates. Their names are misleading, but the text is clear. e.g., from 4880:

> The validity period of the key.  This is the number of seconds after the key creation time that the key expires.

**Comment by dignifiedquire** [comment]
oh, I must have missed that 😅 thanks

File: src/packet/signature/de.rs [hunk 1]
```diff
@@ -38,6 +38,11 @@ fn dt_from_timestamp(ts: u32) -> Option<DateTime<Utc>> {
         .map(|ts| DateTime::<Utc>::from_naive_utc_and_offset(ts, Utc))
 }
 
+/// Convert a u32 to a `Duration`
+fn duration_from_timestamp(ts: u32) -> Option<Duration> {
+    Some(Duration::seconds(ts as i64))
```
Comment (dignifiedquire) [inline src/packet/signature/de.rs:43]: please use `i64::from` instead

**Comment by hko-s** [comment]
> oh, I must have missed that 😅 thanks

The naming is utterly confusing

File: src/packet/signature/de.rs (hunk 1 above)
Comment (hko-s) [reply src/packet/signature/de.rs]: i changed it

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**